### PR TITLE
Importing a BOM works even when it has the same components in dep mgmt

### PR DESCRIPTION
**Pull request by jglick**
Does not correspond to any known bug, but it seems like a potentially unusual practice deserving of test coverage.

CC @stephenc who wondered whether this actually works

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/IntegrationTestSuite.java [hunk 1]
```diff
@@ -654,6 +654,7 @@ public static Test suite()
         suite.addTestSuite( MavenITmng0249ResolveDepsFromReactorTest.class );
         suite.addTestSuite( MavenITmng0187CollectedProjectsTest.class );
         suite.addTestSuite( MavenITmng0095ReactorFailureBehaviorTest.class );
+        suite.addTestSuite( MavenIT0199CyclicImportScopeTest.class );
```
Comment (jglick) [inline core-it-suite/src/test/java/org/apache/maven/it/IntegrationTestSuite.java]: Was unclear what the naming scheme was here, so I just made something up.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenIT0199CyclicImportScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            build("v" + i + "/main", "bundle/target/bundle-" + i + ".jar");
+        }
+    }
+
+    private void build(String module, String expectedArtifact) throws Exception {
+        File testDir = ResourceExtractor.simpleExtractResources(getClass(), "/cyclic-import-scope");
+        Verifier verifier = newVerifier(new File(testDir.getAbsolutePath(), module).getAbsolutePath());
+        verifier.setAutoclean(false);
+        verifier.deleteDirectory("target");
+        verifier.executeGoal("install");
```
Comment (jglick) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenIT0199CyclicImportScopeTest.java]: Tested iteratively using

```sh
rm -rfv repo/test
mvn clean test -f core-it-suite -Prun-its -Dmaven.repo.local=`pwd`/repo -Dtest=org.apache.maven.it.MavenIT0199CyclicImportScopeTest
```

File: core-it-suite/src/test/java/org/apache/maven/it/MavenIT0199CyclicImportScopeTest.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ * under the License.
+ */
+
+import java.io.File;
+import org.apache.maven.it.util.ResourceExtractor;
+
+public class MavenIT0199CyclicImportScopeTest extends AbstractMavenIntegrationTestCase {
+
+    public MavenIT0199CyclicImportScopeTest() {
+        super(ALL_MAVEN_VERSIONS);
```
Comment (jglick) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenIT0199CyclicImportScopeTest.java]: Well, tested against 3.5.2. Did not try running against, say, 2.x. Is there some automated process which does this?

**Comment by jtnord** [comment]
@jglick was there a Jira filed for this - I think this has got lost on the Apache side?

//cc @aheritier

**Comment by aheritier** [comment]
Effectively it's often hard to monitor all PRs in all Maven related repo
I am not sure if someone saw it ( cc @hboutemy @rfscholte ) without a Jira task in the ASF side

**Comment by jglick** [comment]
I do not recall any Apache JIRA issue.

**Comment by rfscholte** [comment]
I've missed a lot of github activity in the period there were read only mirrors. Having 2 overlapping systems (github vs jira) generates a lot of email traffic, I'm not able to keep up and respond to it.

**Comment by jglick** [comment]
@gnodet @cstamas et al., is there someone to review this sort of thing?

**Comment by olamy** [comment]
this looks to be a good idea to have an IT to ensure this will keep working and as is being a documented `feature` :)
Currently this repo doesn't have any build by itself as it's running after a build maven core so we do not have any way to validate changes.
So I have created this #172 to be able to validate such changes. 
once #172 is merged if you could rebase that would be great. 
thanks

**Comment by olamy** [comment]
@jglick the PR has been merged. Can you update your branch? Then I will merge it. Thanks

**Pull request merged**